### PR TITLE
Add first draft of a roadmap document

### PR DESCRIPTION
**Pull request by djhoese**
Addresses the topics presented in https://github.com/vispy/vispy/issues/1987.

A lot of this is my opinion and where *I* think VisPy is going based on my discussions with others and my experience maintaining VisPy for the last couple years. It being my opinion means it might not be accurate or represent what the rest of the @vispy/core developers think.

So everyone please review when you have the time.

### CODE REVIEW

File: roadmap.rst [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+Primitive Visuals
+^^^^^^^^^^^^^^^^^
+
+Along the same lines of preventing low-level APIs leaking into higher levels,
+we'd like to define a set of "primitive" Visual objects. These primitives
+would define a set of basic functionality and that interact the closest with
+the low-level OpenGL layers of VisPy. Using these primitives, users should be
+able to easily create their own, more complex, visualizations without ever
+needing to know the complexities of the underlying laters.
```
Comment (kmuehlbauer) [inline roadmap.rst]: Typo, should be `layers`?

File: roadmap.rst [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+See the `VisPy Wiki <https://github.com/vispy/vispy/wiki/Primitive-Visuals>`_
+for our attempt at defining these types of primities.
```
Comment (kmuehlbauer) [inline roadmap.rst]: Typo, should be `primitives`.

**Review by kmuehlbauer — COMMENTED** [review]
@djhoese This was a fun read! Very nice! I spotted two typo's. Surely have missed others.

File: roadmap.rst (hunk 1 above)
Comment (djhoese) [reply roadmap.rst]: Fixed thanks.

**Review by rossant — APPROVED** [review]
LGTM! I basically agree with everything.

**Comment by djhoese** [comment]
I've heard no complaints so I'm going to merge this. Thanks for the reviews.

**Pull request merged**